README.md

```diff
@@ -1,9 +1,259 @@
-netx.ws
-=======
+# netx.ws
 
+[![Gitter](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/kaazing/gateway?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 [![Build Status][build-status-image]][build-status]
 
 [build-status-image]: https://travis-ci.org/kaazing/netx.ws.svg?branch=develop
 [build-status]: https://travis-ci.org/kaazing/netx.ws
 
-WebSocket RFC-6455 implementation
+netx.ws is a Java-based implementation of [WebSocket RFC 6455](https://tools.ietf.org/html/rfc6455) client with following
+important characteristics:
+
+* Target JVM Version - Java SE 1.6
+* `java.net.Socket` and `java.net.URLConnection` style APIs
+* API for streaming binary and text payloads across WebSocket frames
+* API for sending and receiving text and binary messages
```

Comment:
That's right. I don't have a section in there describing how to use MessageReader/MessageWriter as the API and the implementation is going to change in the next/current sprint.